Angular 17 support

danisss9 opened the pull request:
Still requires fix of:

scripts folder
tests
docs website generation
replace tslint with eslint

piotrdz commented on KNOWN_ISSUES.md:
```diff
@@ -29,3 +29,10 @@ Unfortunately, there is no easy fix for this, other than re-writing all of the l
 
+## Using CSS transform: scale makes slider render incorrectly 
+
+Using `transform : scale(N)` in css will mess with the calculation of the slider. You can workaround this by setting the scale in ngx-slider options as the opposite value of the scale set in css. [See Issue.](https://github.com/danisss9/ngx-slider/issues/2)
```
This probably needs to be entered and linked as an issue in the main repository.

piotrdz commented on src/ngx-slider/lib/slider.component.ts:
```diff
@@ -861,7 +952,7 @@ export class SliderComponent implements OnInit, AfterViewInit, OnChanges, OnDest
     this.updateAriaLabel();
     this.calculateViewDimensions();
-    this.refocusPointerIfNeeded();
+    // this.refocusPointerIfNeeded();
```
@danisss9 Can you follow up on this one? If it really solves a problem, then it should be removed, not commented out.

danisss9 replied:
I don't remember what the issue was, seems to have been from this commit: [prevent infinite loop detectChanges -> ngOnChanges -> detectChanges](https://github.com/danisss9/ngx-slider/commit/c096d47988e63cf3618075a2cfb443ffd4b1b8a5)

We should try uncomment and retest it.

piotrdz replied:
Also resolving this one. I have a rewrite of this function on ng17 branch.

piotrdz reviewed (commented):
Thanks, this looks much better than the previous PR.

I still need to check out the code and run a few checks on the repository but overall, it looks good.

danisss9 commented:
Done a few updates:

1. Updated angular and bootstrap packages to most recent minor version
2. Replaced tslint with eslint (still needs to ajust some rules)
3. Updated tsconfig with the recomended settings for angular 17 (had to disable strict tought)
4. Added escape of all "@" characters when generating Demo App snippets (because of Angular's new control flow syntax)

piotrdz commented on package.json:
```diff
@@ -37,51 +37,54 @@
+    "@types/node": "^20.11.13",
+    "@typescript-eslint/eslint-plugin": "6.19.0",
+    "@typescript-eslint/parser": "6.19.0",
+    "codelyzer": "^6.0.2",
```
It looks like `codelyzer` still pulls in tslint as a dependency. I don't think `codelyzer` is required in any case, so I think it can be removed.

piotrdz replied:
Never mind, I have a solution for it on follow-up branch `ng17`.

piotrdz commented on package.json:
```diff
@@ -37,51 +37,54 @@
+    "ts-node": "~10.9.2",
+    "typedoc": "~0.25.7",
+    "typescript": "~5.2.2",
+    "webdriver-manager": "^12.1.9"
```
`webdriver-manager` pulls in deprecated `requests` and `uuid`. But it is probably fine, as it is only for e2e tests.

piotrdz replied:
Never mind, I have a solution for it on follow-up branch `ng17`.

piotrdz commented on CHANGELOG.md:
```diff
@@ -1,3 +1,22 @@
+# 17.0.0
+ * add support for Angular 17
```
In this pull request, we are only pulling in the work to support Angular 17, the other releases are not ported yet.

I would not want to create a confusing history of changes in this file, so for now, I would roll back the changes and only mention Angular 17.

@danisss9: Can you push this change to the pull request please? Once this is done, I will merge the pull request and then pull in the rest of my changes from ng17 branch.

danisss9 replied:
done

piotrdz reviewed (approved):
@danisss9 Thanks for all your work. I'm happy to merge now.
